src/main/java/org/cirdles/topsoil/dataset/DatasetResource.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2015 CIRDLES.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.cirdles.topsoil.dataset;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.cirdles.topsoil.app.dataset.writer.TSVDatasetWriter;
+
+/**
+ *
+ * @author parizotclement
+ */
+public class DatasetResource {
+
+    private static final Logger LOGGER
+            = Logger.getLogger(DatasetResource.class.getName());
+
+    private final Dataset dataset;
+    private Path path;
+    private final Path basePath;
+
+    public DatasetResource(Dataset dataset, Path basePath, Path path) {
+
+        this.dataset = dataset;
+        this.basePath = basePath;
+        this.path = path;
+    }
+
+    public Dataset getDataset() {
+        return dataset;
+    }
+
+    public void delete() {
+        try {
+            Files.delete(path);
+        } catch (IOException ex) {
+            LOGGER.log(Level.SEVERE, null, ex);
+        }
+    }
+
+    public void save() {
+
+        try {
+            new TSVDatasetWriter().write(dataset, this.path);
+        } catch (IOException ex) {
+            LOGGER.log(Level.SEVERE, null, ex);
+        }
+    }
+
+    public void save(Path path) {
+
+        this.path = path;
+
+        try {
+            new TSVDatasetWriter().write(dataset, this.path);
+        } catch (IOException ex) {
+            LOGGER.log(Level.SEVERE, null, ex);
+        }
+    }
+
+    Path getOpenPath() {
+        return basePath.resolve("open");
+    }
+
+    Path getClosedPath() {
+        return basePath.resolve("closed");
+    }
+
+    public boolean isOpen() {
+        return Optional.ofNullable(path)
+                .map(Path::getParent)
+                .map(parentPath -> parentPath.equals(getOpenPath()))
+                .orElse(false);
+    }
+
+    public boolean isClosed() {
+        return Optional.ofNullable(path)
+                .map(Path::getParent)
+                .map(parentPath -> parentPath.equals(getClosedPath()))
+                .orElse(false);
+    }
+
+    public void open() {
+        if (isOpen()) {
+            return;
+        }
+        try {
```

Comment:
Put a blank line here (between 103 and 104) and below in `close`, please. It's hard to read without.